src/shared/stores/data-value-store.js

```diff
@@ -35,3 +38,7 @@ export const useValueStore = create((set, get) => ({
         return !!dataValue?.comment
     },
 }))
+
+export function useValueStore(selector) {
+    return useStore(valueStore, selector)
+}
```

Comment:
You don't actually have to do this, even though it's fine, it's not necessary - you can call `setState` and `getState` on the returned hook as well.

```js
export const useValueStore = create(....)

useValueStore.setState()
```

You could also use probably do something like this to not have to import `initialState` in tests:
```js

describe('data value store', () => {
    const initialState = useValueStore.getState()
    beforeEach(() => {
        useValueStore.setState(initialState)
    })
```

Reply:
Ah, that's great! How did you figure that out? I couldn't find an api reference in the docs

Reply:
> Ah, that's great! How did you figure that out? I couldn't find an api reference in the docs

A combination of looking at the implementation of the "automagic" testing-link:
import actualCreate from 'zustand' 
```js
const create = (createState) => {
  const store = actualCreate(createState) // this is using default `create` from zustand which is the same for hooks
  const initialState = store.getState()
  storeResetFns.add(() => store.setState(initialState, true))
  return store
}
```

That made me just check the type returned:
![image](https://user-images.githubusercontent.com/13852438/205911916-6b36bd79-1a74-47fb-a0f5-76f8b454ced1.png)